cdap-app-fabric/src/main/java/co/cask/cdap/app/queue/QueueSpecificationGenerator.java

```diff
@@ -73,7 +85,8 @@ public boolean equals(Object o) {
         return false;
       }
       Node other = (Node) o;
```

Comment:
This is not from your PR but this is directly casting an Object to Node without checking if its an instance of it or not first. This equals is wrong. Can you please correct it.